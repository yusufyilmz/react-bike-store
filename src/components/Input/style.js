
import styled from 'styled-components';


export const InputWrapper = styled.input`
    width: ${props => props.width ? props.width+  'px' : '150px'};
    margin-right:20px;
    cursor:pointer;
    border-radius: 5px;
    height:30px;
    font-size: 15px!important;
    margin-top: 20px;
    float: ${props => props.float && props.float}
`;

