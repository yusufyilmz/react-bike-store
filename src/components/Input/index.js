import React from 'react';
import { InputWrapper } from './style';



const Input = ({ type, name, text, onChange, value, onClick, required, placeHolder, width, ...props }) => {
    return (
        <div>
            <label htmlFor={name} >{text}</label>
            <InputWrapper 
               {...props}
                width={width}
                name={name}
                placeholder={placeHolder}
                autoComplete="off"
                id={name}
                type={type}
                onChange={onChange}
                onClick={onClick}
                // required={required}
                value={value}>
            </InputWrapper>
        </div>
    )
}

export default Input;

