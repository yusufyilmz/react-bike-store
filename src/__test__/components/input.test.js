

import React from 'react';
import Input from '../../components/Input';
import { shallow } from 'enzyme'


describe('Input component', () => {

    let value, onChange, wrapper;

    beforeEach(() => {
        value = 'test';
        onChange = jest.fn();
        wrapper = shallow(
            <Input text="test" type='text' onChange={onChange} value={value} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('renders onclick correctly', () => {
        wrapper.find('Input').simulate('change');
        expect(onChange.mock.calls.length).toEqual(1);

    });
})
