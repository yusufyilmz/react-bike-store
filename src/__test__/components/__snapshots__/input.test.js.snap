// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders element correctly 1`] = `
<div>
  <label>
    test
  </label>
  <Input
    autoComplete="off"
    onChange={[MockFunction]}
    type="text"
    value="test"
  />
</div>
`;
